docs/proposals/multiple-sources-for-applications.md

```diff
@@ -0,0 +1,225 @@
+            targetRevision: main
+            valueFiles:
+              - values.yaml
+      chart: mysql
+```
+
+### Combined Example Application yaml
+
+```yaml
+apiVersion: argoproj.io/v1alpha1
+kind: Application
+metadata:
+  name: grafana
+  namespace: argocd
+spec:
+  destination:
+    namespace: monitoring
+    server: https://some.k8s.url.com:6443
+  project: default
+  source:
+    repoURL: https://charts.bitnami.com/bitnami
+    targetRevision: 8.5.8
+    helm:
+    valueFiles:
+        - values.yaml
+    chart: mysql
+  sources:                                          # new field
+    - repoURL: https://charts.bitnami.com/bitnami
+      targetRevision: 8.5.8
+      helm:
+        valueFiles:
+          - values.yaml
+        externalValueFiles:                         # new field
+          - repoURL: https://github.com/KaiReichart/argo-test-values.git
+            targetRevision: main
+            valueFiles:
+              - values.yaml
+      chart: mysql
+    - repoURL: https://charts.bitnami.com/bitnami1
+      targetRevision: 8.5.8
+      helm:
+        valueFiles:
+          - values.yaml
+        externalValueFiles:
+          - repoURL: https://github.com/KaiReichart/argo-test-values.git
+            targetRevision: main
+            valueFiles:
+              - values.yaml
+      chart: mysql
+  syncPolicy:
+    automated: {}
+```
+
+### Use cases
+
+Add a list of detailed use cases this enhancement intends to take care of.
+
+#### Use case 1: 
+As a user, I have an Application that uses the [elasticsearch](https://github.com/helm/charts/tree/master/incubator/elasticsearch) helm chart as source. Today, user needs to create a separate Application to configure the [elasticsearch-exporter](https://github.com/helm/charts/tree/master/stable/elasticsearch-exporter
+) to expose Prometheus metrics.
+https://github.com/argoproj/argo-cd/issues/677
+
+#### Use case 2: 
+We have a Helm Chart which is used in 30+ Services and each of them is customized for 3 possible environments.
+Replicating this Chart 30 times without a centralized Repo looks dirty. Can be a show stopper for the whole migration.
+Modifying the Applica`tion definition is not an option since the whole goal is to reduce the rights that the CI-solution has. Giving it the right to update all Application-definitions from various teams in the argocd namespace is a a hard thing to convince people with.
+
+
+### Implementation Details
+
+#### Attach multiple sources to the Application
+
+To allow multiple sources to the Application, we would add a new field `sources` which would allow users to input list of `ApplicationSource`s. As part of this update and to support backward compatibilty, we would mark field `source` as deprecated and remove it in later revisions.
+
+To avoid complexity on the deciding the list of sources, if both `source` and `sources` fields are specified, we would override the source under `source` field with all the sources under `sources` field.
```

Comment:
To rephrase: if there is at least one source in `sources`, Argo CD will ignore the `source` field. Is that correct?

Should we communicate the misconfiguration in any way (UI blurb, log a warning, etc.)?

Reply:
Yes. That is correct. We will ignore the `source` field if even one resource is added under `sources` field. 

Regarding communication, it would make sense to log warnings once we have UI and cli changes in place. 

**One other way** I can think of using is, making `source` and `sources` field mutually exclusive, it that helps. If user provides both `source` and `sources` field, we give an `invalid yaml` error message. WDYT?

Reply:
I like the idea of being pretty aggressive about rejecting the ambiguous Application. But I don't feel very strongly about that vs. just logging a warning.